aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -264,10 +266,10 @@ impl AptosVM {
         &self,
         executor_view: &'r R,
     ) -> StorageAdapter<'r, R> {
-        StorageAdapter::from_borrowed_with_config(
+        StorageAdapter::new_with_config(
             executor_view,
-            self.0.get_gas_feature_version(),
-            self.0.get_features(),
+            self.vm_impl.get_gas_feature_version(),
```

Comment:
These both should be bundled in a type with defined API to pass between components. At least a note for later refactoring.

Reply:
Sounds good! Let's do it in a later PR